### ISSUE TITLE
Spike: Outline valuable areas to implement testing

### ISSUE THREAD

**Issue by aboydnw** [NASA-IMPACT/nasa-apt#528]
## Description
Goal of this spike to determine where we want to add unit and integration tests.
Do we want to run tests on every PR? Just when we merge something? Other?

## Acceptance Criteria
- [ ] In collaboration with IMPACT and the team, determine areas most valuable to implement testing
- [ ] Outline plan for implementing testing and any details necessary to implementation tickets

**Comment by oliverroick** [comment]
I suggest, we start with these:

- Create new document
- View document
- Edit individual section
	- Edit and save Identifying information
	- Contact information
		- Add new contact
		- Delete contact
		- Edit contact + save
	- References
		- Import from BibTex file
		- Add new reference
		- Edit reference
		- Delete reference
		- Select reference and save
	- Edit and save introduction
	- Edit and save algorithm description
		- Add new input var
		- Delete input var
		- Edit input var
		- Add new output var
		- Delete output var
		- Edit output var
	- Edit and save algorithm usage
	- Edit and save algorithm implementation
	- Edit and save algorithm closeout
- Save & continue flow stepping through the individual steps
- Switching sections
- Comments
	- Create new comment
	- Edit comment
	- Reply to comment
- Contacts
	- Create contacts
	- Edit contacts
- Publish workflow
	- Request review
	- Approve review request + select reviewers
	- Conclude review
	- Request publication
	- Approve publication request
	- Publish document
- Create new version
	- Create minor version
	- Create major version
- Download PDF

NOTE: This is a first overview of the flows we should consider testing. Implementing all of this is a lot of work and likely won't all fit into this quarter. We should consider focusing on high-level aspects (for example not including things like adding/editing input vars) at this point and add details later.

**Comment by aboydnw** [comment]
Thanks @oliverroick 
I like the list. Given that it will take a while to implement, I took a stab at narrowing down what I think may be the most important things to test. I also added a few for testing adding equations, images, and charts, if that's possible. 

I highlighted the ones I think are most important here: https://docs.google.com/document/d/1LknjE_xSOSYmoWKj4RJPkP4hFIiQQJEZI4uu1geodgs/edit?usp=sharing

But I am definitely not married to these, especially considering I'm not as familiar with how challenging each of these will be. Let me know what you think
